src/include/ceph_assert.h

```diff
@@ -112,12 +92,21 @@ using namespace ceph;
 #define ceph_abort_msgf(...)                                             \
   __ceph_abortf( __FILE__, __LINE__, __CEPH_ASSERT_FUNCTION, __VA_ARGS__)
 
+#ifdef __SANITIZE_ADDRESS__
+#define ceph_assert(expr)                           \
```

Comment:
it looks like this change needs to cover `ceph_assert_always()` as well - i'm still seeing some linker errors coming from the `ceph::crypto::Digest` stuff in `common/ceph_crypto.h`:
```
`_ZZN4ceph6crypto6Digest6UpdateEPKhmE19__PRETTY_FUNCTION__' referenced in section `.data.rel' of ../lib/libcommon.a(buffer.cc.o): defined in discarded section `.rodata._ZZN4
ceph6crypto6Digest6UpdateEPKhmE19__PRETTY_FUNCTION__[_ZZN4ceph6crypto6Digest6UpdateEPKhmE15assert_data_ctx]' of ../lib/libcommon.a(buffer.cc.o)                             
`_ZZN4ceph6crypto6Digest7RestartEvE19__PRETTY_FUNCTION__' referenced in section `.data.rel' of ../lib/libcommon.a(buffer.cc.o): defined in discarded section `.rodata._ZZN4ce
ph6crypto6Digest7RestartEvE19__PRETTY_FUNCTION__[_ZZN4ceph6crypto6Digest7RestartEvE15assert_data_ctx]' of ../lib/libcommon.a(buffer.cc.o)
`_ZZN4ceph6crypto6DigestC4E9SECOidTagmE19__PRETTY_FUNCTION__' referenced in section `.data.rel' of ../lib/libcommon.a(buffer.cc.o): defined in discarded section `.rodata._ZZ
N4ceph6crypto6DigestC4E9SECOidTagmE19__PRETTY_FUNCTION__[_ZZN4ceph6crypto6DigestC4E9SECOidTagmE15assert_data_ctx]' of ../lib/libcommon.a(buffer.cc.o)
collect2: error: ld returned 1 exit status
gmake[3]: *** [src/CMakeFiles/ceph-osd.dir/build.make:134: bin/ceph-osd] Error 1
```